src/talk.ts

```diff
@@ -1,5 +1,8 @@
 import { playAudioFile, generateAudio } from './depedenciesLibrary/voice'
 import { llamaInvoke } from './depedenciesLibrary/llm';
+import { Mutex } from 'async-mutex';
+
+const mutex = new Mutex();
```

Comment:
Let's avoid adding packages - mutex likely can be re-implemented easily

Reply:
Aye, will implement this. I don't want to use the existing booleans based mutex since they are really not thread safe.

https://github.com/DirtyHairy/async-mutex/blob/master/src/Semaphore.ts

Reply:
Implemented new one in `mutex.ts`.